R/steps_and_checks.R

```diff
@@ -5,24 +5,41 @@
 #' @param subclass A character string for the resulting class. For example,
 #'   if `subclass = "blah"` the step object that is returned has class
 #'   `step_blah` or `check_blah` depending on the context.
-#' @param ... All arguments to the operator that should be returned.
+#' @param ... All arguments to the operator that should be returned. Required
+#' arguments are `trained`, `skip`, and `id`.
 #' @param .prefix Prefix to the subclass created.
 #'
 #' @seealso [developer_functions]
 #'
 #' @keywords internal
 #' @return An updated step or check with the new class.
 #' @export
-step <- function(subclass, ..., .prefix = "step_") {
-  structure(list(...),
+step <- function(subclass, ..., .prefix = "step_",
+                 call = rlang::caller_env()) {
+  args <- list(...)
+
+  check_string(subclass, call = call)
+  .prefix <- rlang::arg_match0(.prefix, c("step_", "check_", ""),
+                               error_call = call)
+  check_step_check_args(args, call = call)
+
+  structure(args,
     class = c(paste0(.prefix, subclass), "step")
   )
 }
 
 #' @rdname step
 #' @export
-check <- function(subclass, ..., .prefix = "check_") {
-  structure(list(...),
+check <- function(subclass, ..., .prefix = "check_",
+                  call = rlang::caller_env()) {
```

Comment:
```suggestion
check <- function(subclass, ..., .prefix = "check_",
                  call = rlang::caller_env(2)) {
```

gives a better message

<img width="602" alt="Screenshot 2024-10-24 at 12 46 43 PM" src="https://github.com/user-attachments/assets/21bfa92e-849c-466d-b9e2-483ca2410d79">